src/framework/platform/platformwindow.cpp

```
@@ -182,3 +183,68 @@ void PlatformWindow::fireKeysPress()
+            pixelsReflection[realHeightPixels + 4 * j + 1] = pixelsReverse[realHeightPixels + 4 * widthLength - (4 * j + 3)];
+            pixelsReflection[realHeightPixels + 4 * j + 2] = pixelsReverse[realHeightPixels + 4 * widthLength - (4 * j + 2)];
+            pixelsReflection[realHeightPixels + 4 * j + 3] = pixelsReverse[realHeightPixels + 4 * widthLength - (4 * j + 1)];
+
+            pixelsReflection[realHeightPixels + 4 * widthLength - 4 * j] = pixelsReverse[realHeightPixels + 4 * j];
+            pixelsReflection[realHeightPixels + 4 * widthLength - (4 * j + 1)] = pixelsReverse[realHeightPixels + 4 * j + 3];
+            pixelsReflection[realHeightPixels + 4 * widthLength - (4 * j + 2)] = pixelsReverse[realHeightPixels + 4 * j + 2];
+            pixelsReflection[realHeightPixels + 4 * widthLength - (4 * j + 3)] = pixelsReverse[realHeightPixels + 4 * j + 1];
+        }
+        if (widthLength % 2 == 1) {
+            pixelsReflection[realHeightPixels + 1 + 4 * reflectionWidth] = pixelsReverse[realHeightPixels + 1 + 4 * reflectionWidth];
+            pixelsReflection[realHeightPixels + 1 + 4 * reflectionWidth + 1] = pixelsReverse[realHeightPixels + 1 + 4 * reflectionWidth + 3];
+            pixelsReflection[realHeightPixels + 1 + 4 * reflectionWidth + 2] = pixelsReverse[realHeightPixels + 1 + 4 * reflectionWidth + 2];
+            pixelsReflection[realHeightPixels + 1 + 4 * reflectionWidth + 3] = pixelsReverse[realHeightPixels + 1 + 4 * reflectionWidth + 1];
+        }
+    }
+
+    ImagePtr image(new Image(Size(getWidth(), getHeight()), 4, pixelsReflection));
+
+    delete[] pixels;
+    delete[] pixelsReverse;
+
+    image->savePNG(file);
+
+    delete[] pixelsReflection;
+}
+
+void PlatformWindow::makeScreenShot(std::string file) {
+    boost::filesystem::path dirPath(file);
+
+    if (!g_resources.directoryExists(dirPath.parent_path().string())) {
+        return;
```

Error logging?

---

This lines checks is that path exist.
It's prevents crash, when dir doesn't exist.

.parent_path() change:
from: /files/screen1.png
to: /files

---

But there is not feedback/error logged to console when path doesn't exist. So the function silently does nothing?

---

Added error log when path doesn't exist.